pkg/cmd/clients.go

```diff
@@ -130,68 +134,165 @@ func (cc *ClientCmd) CreateClientCmd() *cobra.Command {
+				return errors.WithStack(err)
 			}
-			outType := outputType(cmd.Flags())
-			if err := cc.Out.Render("create"+cmd.Name(), outType, createdClient); err != nil {
-				return errors.Wrap(err, fmt.Sprintf(output.FailedToOutPutInFormat, "mobile client", outType))
+
+			instParams := &ServiceParams{}
+
+			if err := json.Unmarshal(clusterServicePlan.Spec.ServiceInstanceCreateParameterSchema.Raw, instParams); err != nil {
+				return errors.WithStack(err)
+			}
+
+			validServiceName := clusterServiceClass.Spec.ExternalName
+			extMeta := clusterServiceClass.Spec.ExternalMetadata.Raw
+			var extServiceClass ExternalServiceMetaData
+			if err := json.Unmarshal(extMeta, &extServiceClass); err != nil {
+				return errors.WithStack(err)
+			}
+
+			si := v1beta1.ServiceInstance{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "servicecatalog.k8s.io/v1beta1",
+					Kind:       "ServiceInstance",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:    namespace,
+					GenerateName: validServiceName + "-",
+				},
+				Spec: v1beta1.ServiceInstanceSpec{
+					PlanReference: v1beta1.PlanReference{
+						ClusterServiceClassExternalName: clusterServiceClass.Spec.ExternalName,
+					},
+					ClusterServiceClassRef: &v1beta1.ClusterObjectReference{
+						Name: clusterServiceClass.Name,
+					},
+					ClusterServicePlanRef: &v1beta1.ClusterObjectReference{
+						Name: "default",
+					},
+					ParametersFrom: []v1beta1.ParametersFromSource{
+						{
+							SecretKeyRef: &v1beta1.SecretKeyReference{
+								Name: clientId + "-apb-" + "params",
+								Key:  "parameters"},
+						},
+					},
+				},
+			}
+
+			if _, err := cc.scClient.ServicecatalogV1beta1().ServiceInstances(namespace).Create(&si); err != nil {
+				return errors.WithStack(err)
+			}
+			fmt.Println("creating service")
+			pSecret := v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clientId + "-apb-" + "params",
+				},
+			}
+			pSecret.Data = map[string][]byte{}
+
+			parameters := map[string]string{}
+			parameters["appName"] = name
+			parameters["appIdentifier"] = appIdentifier
+			secretData, err := json.Marshal(parameters)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			pSecret.Data["parameters"] = secretData
+			if _, err := cc.k8Client.CoreV1().Secrets(namespace).Create(&pSecret); err != nil {
+				return errors.WithStack(err)
+			}
+
+			noWait, err := cmd.PersistentFlags().GetBool("no-wait")
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			if noWait {
+				return nil
+			}
+			timeout := int64(5 * 60) // ten minutes
+			w, err := cc.scClient.ServicecatalogV1beta1().ServiceInstances(namespace).Watch(metav1.ListOptions{TimeoutSeconds: &timeout})
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			for {
+				select {
+				case msg, ok := <-w.ResultChan():
+					if !ok {
+						fmt.Println("Timedout waiting. It seems to be taking a long time for the service to provision. Your service may still be provisioning.")
```

Comment:
I think we should make this error call less about a service (I know we are provisioning a service instance) and say timedout creating the mobile client